online-survey-app/src/app/tangy-forms/tangy-forms-player/tangy-forms-player.component.ts

```
@@ -52,6 +58,32 @@ export class TangyFormsPlayerComponent implements OnInit {
   async ngOnInit(): Promise<any> {
     this.window = window;
 
+    // we are using the query parameters to get the actor and auth information
+    this.route.queryParamMap.subscribe((query) => {
+      const actorRaw = query.get('actor');
+      const authRaw = query.get('auth');
+
+      this.xapiEndpoint = query.get('endpoint');
+      this.xapiRegistration = query.get('registration');
+
+      // here we parse the auth and actor query parameter
+      try {
+        this.xapiAuth = JSON.parse(authRaw)[0];
+        const xapiActorData = JSON.parse(actorRaw);
+        this.xapiActor = new Agent(
+          xapiActorData.name,
+          xapiActorData.mbox,
+          xapiActorData.mbox_sha1sum,
+          xapiActorData.openid,
+          xapiActorData.account
+        );
+
+      } catch (e) {
+        console.warn('Error parsing auth or actor query parameters:', e);
```

Exceptions should not be caught and swallowed. 

It is acceptable if passing an invalid parameter causes a crash / exception or error message. 

If the exception is swallowed and we just allow things to continue as if nothing had gone wrong, then data won't be sent to the xAPI endpoint, and users might just wonder why the data isn't handled as expected.

---

Now am showing an UI message 'This form was opened with incomplete link information. You can continue, but your progress might not be saved properly.'.